File: src/main/java/com/example/BookCatalogueSpringBootWebApp/service/BookService.java
package com.example.BookCatalogueSpringBootWebApp.service;

import org.springframework.stereotype.Service;

import com.example.BookCatalogueSpringBootWebApp.model.Book;
import com.example.BookCatalogueSpringBootWebApp.repository.BookRepository;

import java.util.List;
import java.util.Optional;  

@Service
public class BookService {

    private List<Book> books;
    private final BookRepository bookRepository;

    public BookService() {
        bookRepository = new BookRepository();
        this.books = bookRepository.loadBooksFromRepository();
    }

    public List<Book> getAllBooks() {
        return books;
    }

    public Optional<Book> getBookById(Long id) {
        return books.stream().filter(book -> book.getId() == id).findFirst();
    }
    
}

File: frontend/src/service/bookService.js
import axios from 'axios';

const API_URL = '/api/books';

class BookService {

    getAllBooks() {
        return axios.get(API_URL);
    }

    getBook(id) {
        return axios.get(`${API_URL}/${id}`);
    }
}

export default new BookService();